src/ground_vehicle.cpp

```diff
@@ -141,7 +141,7 @@ int GroundVehicle<T, Type>::GetAcceleration() const
 	}
 	/* Air drag; the air drag coefficient is in an arbitrary NewGRF-unit,
 	 * so we need some magic conversion factor. */
-	resistance += (area * this->gcache.cached_air_drag * speed * speed) / 1000;
+	resistance += static_cast<size_t>(area) * this->gcache.cached_air_drag * speed * speed / 1000;
```

Comment:
`resistance` is not a `size_t` but `int64`, so it is still broken for 32 bit builds.

Reply:
awh, and here I thought I checked them all :( Nicely spotted :D